### DISCUSSION TITLE
Support protobuf publishing in Studio

### DISCUSSION THREAD

**Discussion by KerstinKeller** [foxglove/community#719] [Q&A]
It would be great to be able to support publising not only `json`, `cbor` and `ros1`, but also `protobuf` encoding (or any encoding which is support for subscription) from Studio (e.g. from the publish or teleop panel).
As a bonus, it would be great if the user could configure as a setting for that panel, how they want to publish (e.g. they want to publish an image annotation in protobuf, but a log message as json).

**Comment by amacneil** [comment]
Per live discussion - it might be easier to start with support for the `foxglove.*` protos only, so that we don't have to solve the generic problem of how to fetch unknown message schemas from the websocket server.

**Reply by defunctzombie** [reply]
My feeling on this workflow is that we should continue to make publishing easy for custom panels and allow publishing from a custom panel whatever data the data source supports sending to the underlying middlware. I believe this is already possible today with websocket connections if your websocket server accepts the published data from your custom panel.
